### PR TITLE
Add metrics annotations

### PR DESCRIPTION
**Pull request by pipo02mix**
Add Prometheus annotations in the nginx controller deployment and expose the port in the service. 

Also, I cleaned all default backend remaining stuff

### CODE REVIEW

File: helm/kubernetes-nginx-ingress-controller-chart/charts/tempresources/templates/configmap.yaml [hunk 1]
```diff
@@ -48,9 +48,6 @@ data:
           "enabled": false,
         },
       },
-      "defaultBackend": {
-        "name": "{{ .Values.defaultBackend.name }}"
-      },
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/charts/tempresources/templates/configmap.yaml]: Cleaning legacy

File: helm/kubernetes-nginx-ingress-controller-chart/charts/tempresources/values.yaml [hunk 2]
```diff
@@ -35,10 +35,7 @@ controller:
 
   service:
     enabled: false
-
-defaultBackend:
-  name: "temp-default-http-backend"
-
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/charts/tempresources/values.yaml]: More cleaning

File: helm/kubernetes-nginx-ingress-controller-chart/values.yaml [hunk 3]
```diff
@@ -72,28 +72,6 @@ controller:
...
-    limits:
-      cpu: 10m
-      memory: 20Mi
-    requests:
-      cpu: 10m
-      memory: 20Mi
-
-  userID: 33
-  groupID: 33
-
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/values.yaml]: More cleaning

**Review by rossf7 — APPROVED** [review]
LGTM. Much <3 for the cleanup.

**Pull request merged**